src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py

```diff
@@ -334,13 +349,15 @@ def update_container_settings(resource_group_name, name, docker_registry_server_
         settings.append('DOCKER_REGISTRY_SERVER_USERNAME=' + docker_registry_server_user)
     if docker_registry_server_password is not None:
         settings.append('DOCKER_REGISTRY_SERVER_PASSWORD=' + docker_registry_server_password)
-
     if docker_custom_image_name is not None:
-        settings.append('DOCKER_CUSTOM_IMAGE_NAME=' + docker_custom_image_name)
-        _add_linux_fx_version(resource_group_name, name, docker_custom_image_name)
-    update_app_settings(resource_group_name, name, settings, slot)
-    settings = get_app_settings(resource_group_name, name, slot)
-    return _mask_creds_related_appsettings(_filter_for_container_settings(settings))
+        returnVal += str(_add_linux_fx_version(resource_group_name, name, docker_custom_image_name))
+
+    if docker_registry_server_user or docker_registry_server_password or docker_registry_server_url:
+        update_app_settings(resource_group_name, name, settings, slot)
+        settings = get_app_settings(resource_group_name, name, slot)
+        returnVal += str(_mask_creds_related_appsettings(_filter_for_container_settings(settings)))
```

Comment:
Again, we don't need a conversion here.
Also you can add a new field with the value of the `docker_custom_image_name ` to get the information complete